libs/bootstrap-desktop.sh

```diff
@@ -0,0 +1,21 @@
+# Set environment variables for using vendored dependencies in desktop builds.
+#
+# This file should be used via `source ./libs/bootstrap-desktop.sh` and will
+# not have the desired effect if you try to run it directly, because it
+# uses `export` to set environment variables.
+
+if [ ! -f "$(pwd)/libs/build-all.sh" ]; then
+  echo "ERROR: bootstrap-desktop.sh should be run from the root directory of the repo"
+else
+  if [ $(uname -s) == "Darwin" ]; then
+    APPSERVICES_PLATFORM_DIR="$(pwd)/libs/desktop/darwin"
+  else
+    APPSERVICES_PLATFORM_DIR="$(pwd)/libs/desktop/linux-x86-64"
+  fi
+  if [ ! -d "$APPSERVICES_PLATFORM_DIR" ]; then
```

Comment:
Nit: I'm not quite sure about this branch, `build-all.sh` is already smart enough to detect already-built artifacts on a per-lib basis (this predicate will not account for partial builds).

Reply:
On my machine, I found that it indeed skipped the build part, but it would still do the "zip everything up into a tarball" part which took on the order to 10-20 seconds and was super annoying.  But maybe I should investigate how to skip that step in `build-all.sh` rather than work around it here...

Reply:
I gave up on trying to get clever about fixing `build-all.sh`, and just modified the test in my new script to check explicitly for both target directories in the case of partial builds.